pennylane/ops/qutrit/parametric_ops.py

```diff
@@ -306,10 +309,154 @@ def compute_matrix(theta, subspace=(0, 1)):  # pylint: disable=arguments-differ
         mat[slices[0]] = mat[slices[3]] = c
         mat[slices[1]] = -s
         mat[slices[2]] = s
+
         return qml.math.convert_like(mat, theta)
 
     def adjoint(self):
         return TRY(-self.data[0], wires=self.wires, subspace=self.subspace)
 
     def pow(self, z):
         return [TRY(self.data[0] * z, wires=self.wires, subspace=self.subspace)]
+
+
+class TRZ(Operation):
+    r"""The single qutrit Z rotation
+
+    Performs the RZ operation on the specified 2D subspace. The subspace is
+    given as a keyword argument and determines which two of three single-qutrit
+    basis states the operation applies to.
+
+    The construction of this operator is based on section 3 of
+    `Di et al. (2012) <https://arxiv.org/abs/1105.5485>`_.
+
+    .. math:: TR_z^{jk}(\phi) = \exp(-i\phi\sigma_z^{jk}/2),
+                \sigma_z^{jk} = |j\rangle\langle j| - |k\rangle\langle k|,
+                j, k \in \{0, 1, 2\}, j \neq k
+
+    **Details:**
+
+    * Number of wires: 1
+    * Number of parameters: 1
+    * Number of dimensions per parameter: (0,)
+
+    Args:
+        phi (float): rotation angle :math:`\phi`
+        wires (Sequence[int] or int): the wire the operation acts on
+        subspace (Sequence[int]): the 2D subspace on which to apply operation
+        do_queue (bool): Indicates whether the operator should be
+            immediately pushed into the Operator queue (optional)
+        id (str or None): String representing the operation (optional)
+
+    **Example**
+
+    The specified subspace will determine which basis states the operation actually
+    applies to:
+
+    >>> qml.TRZ(0.5, wires=0, subspace=[0, 1]).matrix()
+    array([[0.96891242-0.24740396j, 0.        +0.j        , 0.        +0.j        ],
+           [0.        +0.j        , 0.96891242+0.24740396j, 0.        +0.j        ],
+           [0.        +0.j        , 0.        +0.j        , 1.        +0.j        ]])
+
+    >>> qml.TRZ(0.5, wires=0, subspace=[0, 2]).matrix()
+    array([[0.96891242-0.24740396j, 0.        +0.j        , 0.        +0.j        ],
+           [0.        +0.j        , 1.        +0.j        , 0.        +0.j        ],
+           [0.        +0.j        , 0.        +0.j        , 0.96891242+0.24740396j]])
+
+    >>> qml.TRZ(0.5, wires=0, subspace=[1, 2]).matrix()
+    array([[1.        +0.j        , 0.        +0.j        , 0.        +0.j        ],
+           [0.        +0.j        , 0.96891242-0.24740396j, 0.        +0.j        ],
+           [0.        +0.j        , 0.        +0.j        , 0.96891242+0.24740396j]])
```

Comment:
Examples need to be updated to consider new subspace format, right? (TRY/TRX too)

```suggestion
    >>> qml.TRZ(0.5, wires=0, subspace=(0, 1)).matrix()
    array([[0.96891242-0.24740396j, 0.        +0.j        , 0.        +0.j        ],
           [0.        +0.j        , 0.96891242+0.24740396j, 0.        +0.j        ],
           [0.        +0.j        , 0.        +0.j        , 1.        +0.j        ]])

    >>> qml.TRZ(0.5, wires=0, subspace=(0, 2)).matrix()
    array([[0.96891242-0.24740396j, 0.        +0.j        , 0.        +0.j        ],
           [0.        +0.j        , 1.        +0.j        , 0.        +0.j        ],
           [0.        +0.j        , 0.        +0.j        , 0.96891242+0.24740396j]])

    >>> qml.TRZ(0.5, wires=0, subspace=(1, 2)).matrix()
    array([[1.        +0.j        , 0.        +0.j        , 0.        +0.j        ],
           [0.        +0.j        , 0.96891242-0.24740396j, 0.        +0.j        ],
           [0.        +0.j        , 0.        +0.j        , 0.96891242+0.24740396j]])
```

Reply:
I was thinking about it when I first changed the `TRX` subspace to be a tuple by default, but during subspace validation the given subspace is cast into a tuple anyway, so users can pass in lists without causing anything to break.